python/tank/platform/validation.py

```diff
@@ -641,57 +641,64 @@ def __validate_settings_hook(self, settings_key, schema, hook_name):
 
         hooks_folder = self._tank_api.pipeline_configuration.get_hooks_location()
 
-        # if setting is default, assume everything is fine
-        if hook_name == constants.TANK_BUNDLE_DEFAULT_HOOK_SETTING:
-            # assume that each app contains its correct hooks
-            return
+        hook_paths_to_validate = []
 
-        elif hook_name.startswith("{self}"):
-            # assume that each app contains its correct hooks
-            return
-        
-        elif hook_name.startswith("{config}"):
-            # config hook 
-            path = hook_name.replace("{config}", hooks_folder)
-            hook_path = path.replace("/", os.path.sep)
-
-        elif hook_name.startswith("{engine}"):
-            # engine hook. see if there is a current engine we can use to
-            # validate against. there should be an engine, but in the case
-            # where validation is being run outside of or before engine
-            # startup, continue and assume the hook exists similar to app
-            # hooks.
-            from .engine import current_engine
-            if current_engine():
-                path = os.path.join(current_engine().disk_location, "hooks")
-                hook_path = path.replace("/", os.path.sep)
-            else:
+        for hook_path in hook_value.split(":"):
+
+            # if setting is default, assume everything is fine
+            if hook_path == constants.TANK_BUNDLE_DEFAULT_HOOK_SETTING:
+                # assume that each app contains its correct hooks
                 return
-        
-        elif hook_name.startswith("{$") and "}" in hook_name:
-            # environment variable: {$HOOK_PATH}/path/to/foo.py
-            # lazy (runtime) validation for this - it may be beneficial
-            # not to actually set the environment variable until later
-            # in the life cycle of the engine 
-            return
-        
-        elif hook_name.startswith("{") and "}" in hook_name:
-            # referencing other instances of items
-            # this cannot be easily validated at this point since
-            # no well defined runtime state exists at the time of validation
-            return
 
-        else:
-            # our standard case
-            hook_path = os.path.join(hooks_folder, "%s.py" % hook_name)
+            elif hook_path.startswith("{self}"):
+                # assume that each app contains its correct hooks
+                continue
```

Comment:
This is why the bug never revealed itself. All of our inheritance examples were based on the base hook living in the app which we were assuming existed. So value like `{self}/foobar.py:{config}/foobar.py` was treated as a single path, but never validated.